runtime/gc_vlhgc/VLHGCAccessBarrier.cpp

```diff
@@ -281,12 +284,47 @@ MM_VLHGCAccessBarrier::jniGetPrimitiveArrayCritical(J9VMThread* vmThread, jarray
 		*isCopy = JNI_FALSE;
 	}
 
-	if (alwaysCopyInCritical || !indexableObjectModel->isInlineContiguousArraylet(arrayObject)) {
+	if (alwaysCopyInCritical) {
 		/* alwaysCopyInCritical or discontiguous (including 0 size array) */
 		copyArrayCritical(vmThread, &data, arrayObject, isCopy);
 	} else if (indexableObjectModel->isVirtualLargeObjectHeapEnabled() && !indexableObjectModel->isDataAdjacentToHeader(arrayObject)) {
 		/* off heap enabled and not adjacent */
 		data = (void *)indexableObjectModel->getDataPointerForContiguous(arrayObject);
+	} else if (!indexableObjectModel->isInlineContiguousArraylet(arrayObject)) {
+		/* an array having discontiguous extents is another reason to force the critical section to be a copy */
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
```

Comment:
in the light of the previous comment, this ifdef could be just above 

'else if (!indexableObjectModel->isInlineContiguousArraylet(arrayObject)) {'

 end its maching endif should be just after 
 
 ```		{
 			copyArrayCritical(vmThread, &data, arrayObject, isCopy);
 		}
 ```